test/src/test/java/hudson/model/QueueTest.java

```diff
@@ -1135,4 +1136,55 @@ private void checkCancelOperationUsingUrl(Function<Queue.Item, String> urlProvid
             assertTrue(currentOne.getFuture().isCancelled());
         }
     }
+
+    @Test
+    @Issue("JENKINS-57805")
+    public void brokenAffinityKey() throws Exception {
+        BrokenAffinityKeyProject brokenProject = r.createProject(BrokenAffinityKeyProject.class, "broken-project");
+        // Before the JENKINS-57805 fix, the test times out because the `NullPointerException` repeatedly thrown from
+        // `BrokenAffinityKeyProject.getAffinityKey()` prevents `Queue.maintain()` from completing.
+        r.buildAndAssertSuccess(brokenProject);
+    }
+
+    public static class BrokenAffinityKeyProject extends Project<BrokenAffinityKeyProject, BrokenAffinityKeyBuild> implements TopLevelItem {
+        public BrokenAffinityKeyProject(ItemGroup parent, String name) {
+            super(parent, name);
+        }
+        @Override
+        public String getAffinityKey() {
+            throw new NullPointerException("oops!");
+        }
+        @Override
+        protected Class<BrokenAffinityKeyBuild> getBuildClass() {
+            return BrokenAffinityKeyBuild.class;
+        }
+        @Override
```

Comment:
This override can be deleted.

Reply:
As far as I can tell it is not defined higher up in the type hierarchy, and if I remove it the build fails:

> hudson/model/QueueTest.java:[1149,18] error: BrokenAffinityKeyProject is not abstract and does not override abstract method getDescriptor() in TopLevelItem

Both `FreeStyleProject` and `WorkflowJob` implement the method, maybe you are thinking of some other type with a `getDescriptor` method?

Reply:
Ah yes, `AbstractDescribableImpl`.